src/frame.c

```diff
@@ -535,6 +559,10 @@ void frame_process(frame_t *st, size_t length, logical_channel_t lc)
         if (hdr.hef)
             offset += parse_hef(st->buffer + offset, audio_end - offset, &hef);
         prog = hef.prog_num;
+        avg = calc_avg_packets(&hdr);
+        seq = (hdr.seq - hdr.pfirst) % MAX_AUDIO_PACKETS;
```

Comment:
This calculation looks dangerous. `hdr.seq` and `hdr.pfirst` are unsigned integers. If `hdr.seq` is 0 and `hdr.pfirst` is 1, then an overflow occurs and the result of the subtraction is `UINT_MAX`. The final result is still correct since the integer modulus happens to be a multiple of `MAX_AUDIO_PACKETS`, but I think it would be safer to add `MAX_AUDIO_PACKETS` into the sum to avoid overflow.

```suggestion
        seq = (MAX_AUDIO_PACKETS + hdr.seq - hdr.pfirst) % MAX_AUDIO_PACKETS;
```

Reply:
Yeahh... Good idea. It wouldn't hurt to add that to prevent any weird issues.